### PR TITLE
Global Emotes

### PR DESCRIPTION
**Pull request by binaryoverload**
# Pull request

 - [x] I have read and agreed to the [CODE OF CONDUCT](https://github.com/CascadeBot/CascadeBot/blob/master/CODE_OF_CONDUCT.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [trello](https://trello.com/b/gUQehy4l/bot).
 - [x] I have tested all of my changes.
 - [x] I have followed the style of the rest of the project.
 
## Added/Changed
- Added reading global emotes from config
- Added helper method for easy display of emotes

### CODE REVIEW

File: config.example.yml [hunk 1]
```diff
@@ -62,6 +62,19 @@ security_levels:
   developer:
   staff:
 
+
+# Global emotes the bot can use to display various information
+global_emotes:
```
Comment (weeryan17) [inline config.example.yml]: We should make this a map. i feel like hardcoding this is a bad idea. Like to boils down to a map anyway

File: src/main/java/com/cascadebot/cascadebot/commands/informational/ServerInfoCommand.java [hunk 2]
```diff
@@ -54,6 +54,11 @@ public CommandType getType() {
         return CommandType.INFORMATIONAL;
     }
 
+    @Override
+    public Permission getPermission() {
+        return Permission.SERVER_INFO_COMMAND;
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/informational/ServerInfoCommand.java]: It begins, but yegh why is permissions stuff in this pr

File: src/main/java/com/cascadebot/cascadebot/data/objects/GuildPermissions.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package com.cascadebot.cascadebot.data.objects;
+
+public class GuildPermissions {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/data/objects/GuildPermissions.java]: See previous review

File: src/main/java/com/cascadebot/cascadebot/permissions/Permission.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+package com.cascadebot.cascadebot.permissions;
+
+import com.cascadebot.cascadebot.commandmeta.CommandType;
+import com.cascadebot.cascadebot.utils.CollectionUtils;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Map;
+
+public enum Permission {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/permissions/Permission.java]: Do i even need to say it

File: src/main/java/com/cascadebot/cascadebot/commands/informational/ServerInfoCommand.java (hunk 2 above)
Comment (binaryoverload) [reply src/main/java/com/cascadebot/cascadebot/commands/informational/ServerInfoCommand.java]: Oh 🤔

File: config.example.yml (hunk 1 above)
Comment (binaryoverload) [reply config.example.yml]: Why is it a bad idea? The IDs litterally never change

**Comment by binaryoverload** [comment]
@weeryan17 Permissions is stuck in this for now, I haven't really implemented anything to do with handling permissions so it's fine

File: config.example.yml (hunk 1 above)
Comment (weeryan17) [reply config.example.yml]: I meant it's a bad idea to hardcode the names, not the ids.

Comment (binaryoverload) [reply config.example.yml]: Oh the names aren't hardcoded, they can be literally anything

**Comment by binaryoverload** [comment]
Yeah sorry about that, this is the reason I'm transitioning to branches for stuff like this

**Pull request merged**